### PR TITLE
Set Version in Makefile from src/templates_parser.ads.

### PR DESCRIPTION
**Pull request by asarhaddon**
THis will simplify updates. As long as the duplication is avoided, the
version could of course be propagated in the other direction.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
 .SILENT:
 
-VERSION	= 20.0
+VERSION	:= $(shell sed src/templates_parser.ads -ne \
```
Comment (TurboGit) [inline Makefile]: This will not work on Windows.

File: Makefile [hunk 2]
```diff
@@ -161,6 +161,12 @@ makefile.setup: setup
 setup: tp_xmlada.gpr force
 	printf " $(foreach v,$(ALL_OPTIONS),$(v) = $($(v))\n)" > makefile.setup
 
+build: src/templates_parser-version.adb
+src/templates_parser-version.adb:
+	printf -- \
+'--  Generated by ../Makefile.\nseparate (Templates_Parser)\nfunction Version return String is\nbegin\n   return "$(VERSION)";\nend Version;\n' \
```
Comment (TurboGit) [inline Makefile]: Line should not be longer than 79 characters.

Comment (TurboGit) [reply Makefile]: Apart from that, this looks good to me.

**Comment by asarhaddon** [comment]
I agree, but I have not managed to break the line without loosing the formatting of the installed Ada source, which seems more important than the formatting of the Makefile.

**Comment by TurboGit** [comment]
> I agree, but I have not managed to break the line without loosing the formatting of the installed Ada source, which seems more important than the formatting of the Makefile.

You can certainly have a set to printf or even a set of echo to avoid that:
   echo '-- generated ...' > file
   echo 'function .. ' >> file
etc

**Comment by asarhaddon** [comment]
Ping?

File: Makefile [hunk 3]
```diff
@@ -145,6 +145,14 @@ makefile.setup: setup
 setup: tp_xmlada.gpr force
 	printf " $(foreach v,$(ALL_OPTIONS),$(v) = $($(v))\n)" > makefile.setup
 
+version_path := src/templates_parser-version.adb
+version_data := -- Generated by ../Makefile.\nseparate \
```
Comment (TurboGit) [inline Makefile]: We need two spaces after -- to avoid a compilation warning:
> templates_parser-version.adb:1:04: (style) space required

**Comment by TurboGit** [comment]
Sorry for the late reply. This is ok to me now. Thanks.

**Pull request merged**